### ISSUE TITLE
Thread is waiting for a slower thread.

### ISSUE THREAD

**Issue by pacu** [Electric-Coin-Company/zcash-swift-wallet-sdk#774]
Reproduced with ECC Reference Wallet commit c59257b27c5d085231cbb68cb27067c69b6826b9 on iOS 16.3
````
2023-02-10 14:04:38.936517-0300 ECC Wallet[14179:6078910] [] BlockDownloader[65] - downloadAndStoreBlocks(using:at:maxBlockBufferSize:totalProgressRange:) - DEBUG 🐞 - Downloading blocks in range: 1745348...1745447
2023-02-10 14:04:46.682883-0300 ECC Wallet[14179:6078910] [] BlockValidator[65] - validate() - DEBUG 🐞 - validateChainFinished
Thread Performance Checker: Thread running at QOS_CLASS_USER_INITIATED waiting on a lower QoS thread running at QOS_CLASS_DEFAULT. Investigate ways to avoid priority inversions
PID: 14179, TID: 6078594
Backtrace
=================================================================
3   ECC Wallet                          0x00000001014c1ff0 _ZN3std6thread4park17h499602b8191bfdebE + 60
4   ECC Wallet                          0x00000001015c453c _ZN17crossbeam_channel7context7Context4with28_$u7b$$u7b$closure$u7d$$u7d$17ha462081da899aef3E + 1916
5   ECC Wallet                          0x0000000101449b90 _ZN20zcash_client_backend11welding_rig22scan_block_with_runner17hce3f97efb1d72617E + 9180
6   ECC Wallet                          0x00000001015c6778 _ZN19zcash_client_sqlite5chain21fsblockdb_with_blocks17h2a33c2052ce509daE + 2080
7   ECC Wallet                          0x00000001015cd3d4 _ZN20zcash_client_backend8data_api5chain18scan_cached_blocks17h0766f68055201498E + 10572
8   ECC Wallet                          0x00000001015a5490 zcashlc_scan_blocks + 600
9   ECC Wallet                          0x00000001008064fc $s19ZcashLightClientKit0A11RustBackendC10scanBlocks13fsBlockDbRoot6dbData5limit11networkTypeSb10Foundation3URLV_AKs6UInt32VAA07NetworkQ0OtFZ + 1196
10  ECC Wallet                          0x0000000100808e88 $s19ZcashLightClientKit0A11RustBackendCAA0aeF7WeldingA2aDP10scanBlocks13fsBlockDbRoot6dbData5limit11networkTypeSb10Foundation3URLV_AMs6UInt32VAA07NetworkR0OtFZTW + 16
11  ECC Wallet                          0x000000010072651c $s19ZcashLightClientKit16BlockScannerImplV10scanBlocks2at18totalProgressRange7didScanySNySiG_AHySiYactYaKFTY0_ + 1020
12  libswift_Concurrency.dylib          0x00000001a08c4f6c 28CA7EA7-2BA5-3AE9-9170-6F17DD614EE1 + 266092
13  libswift_Concurrency.dylib          0x00000001a08c5e78 28CA7EA7-2BA5-3AE9-9170-6F17DD614EE1 + 269944
14  libdispatch.dylib                   0x000000010447e468 _dispatch_root_queue_drain + 408
15  libdispatch.dylib                   0x000000010447ee64 _dispatch_worker_thread2 + 196
16  libsystem_pthread.dylib             0x00000001e4704dbc _pthread_wqthread + 228
17  libsystem_pthread.dylib             0x00000001e4704b98 start_wqthread + 8
2023-02-10 14:05:15.438308-0300 ECC Wallet[14179:6078910] [] CompactBlockProcessor[864] - notifyProgress(_:) - DEBUG 🐞 - progress: syncing(ZcashLightClientKit.BlockProgress(startHeight: 1745348, targetHeight: 1979375, progressHeight: 1745352))
2023-02-10 14:05:15.439350-0300 ECC Wallet[14179:6078608] [] BlockScanner[80] - scanBlocks(at:totalProgressRange:didScan:) - DEBUG 🐞 - Scanned 5 blocks in 28.75287890434265 seconds
2023-02-10 14:05:15.439613-0300 ECC Wallet[14179:6078911] [] Home[132] - bindToEnvironmentEvents() - DEBUG 🐞 - --SHOULD NOT SLEEP--
````

**Comment by Chlup** [comment]
This can be reproduced in the Secant app if you tap on `Shield funds` button.

**Comment by true-jared** [comment]
Closing as probably no longer relevant. Reopen if we see it again.

**Comment by tw0po1nt** [comment]
I am seeing this issue and it seems to be completely locking up our app:

```
Thread Performance Checker: Thread running at User-initiated quality-of-service class waiting on a lower QoS thread running at Default quality-of-service class. Investigate ways to avoid priority inversions
PID: 23745, TID: 350183
Backtrace
=================================================================
3   secant-mainnet                      0x0000000103ffbb60 _ZN3std6thread4park17h97e9cc3f69426088E + 56
4   secant-mainnet                      0x00000001041257ac _ZN17crossbeam_channel7context7Context10wait_until17h24479b21811c4873E.17578 + 1044
5   secant-mainnet                      0x00000001041250f4 _ZN17crossbeam_channel7context7Context4with28_$u7b$$u7b$closure$u7d$$u7d$17h0bd43ff86e3ff442E + 128
6   secant-mainnet                      0x00000001040f76ac _ZN20zcash_client_backend8scanning23scan_block_with_runners17h58c960651f900031E + 17772
7   secant-mainnet                      0x00000001042684fc _ZN20zcash_client_backend8data_api5chain18scan_cached_blocks17h8615bb3ab8da043bE + 18668
8   secant-mainnet                      0x00000001041f8528 zcashlc_scan_blocks + 940
9   secant-mainnet                      0x00000001026fabc8 $s19ZcashLightClientKit0A11RustBackendV10scanBlocks10fromHeight0I5State5limitAA11ScanSummaryVs5Int32V_AA04TreeK0Vs6UInt32VtYaKFTY0_ + 2464
10  libswift_Concurrency.dylib          0x000000020a8b410c _ZN5swift34runJobInEstablishedExecutorContextEPNS_3JobE + 324
11  libswift_Concurrency.dylib          0x000000020a8b4fd4 _ZL17swift_job_runImplPN5swift3JobENS_11ExecutorRefE + 80
12  libdispatch.dylib                   0x00000001082a936c _dispatch_root_queue_drain + 372
13  libdispatch.dylib                   0x00000001082a9f18 _dispatch_worker_thread2 + 256
14  libsystem_pthread.dylib             0x0000000106cd37d8 _pthread_wqthread + 224
15  libsystem_pthread.dylib             0x0000000106cd25d4 start_wqthread + 8
```

To reproduce, you should be able to pull [this branch](https://github.com/nighthawk-apps/nighthawk-ios-wallet/tree/feature/tca-upgrade)

1. Fire up the app on a simulator or device.
2. On a fresh launch, the app will present the new wallet screen just fine. You can even create a new wallet.
3. On the create wallet success screen, try to continue to the home screen (which starts the synchronizer)
4. It locks up
5. Kill the app and relaunch
6. The app tries to start the synchronizer right on launch and locks up, leaving the app stuck on the splash screen.

Any assist would be appreciated.
